### PR TITLE
[FLINK-19118][python] Support Expression in the operations of Python Table API

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request support Expression in the operations of Python Table API*

## Brief change log

  - *Update the operations in Table to allow both string and Expression arguments*

## Verifying this change
 
  - Have updated the existing tests to use Expression.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (Will add documentation in a separate PR)

### CODE REVIEW

File: flink-python/pyflink/table/tests/test_pandas_udf.py [hunk 1]
```diff
@@ -58,8 +59,8 @@ def test_basic_functionality(self):
 
         t = self.t_env.from_elements([(1, 2, 3), (2, 5, 6), (3, 1, 9)], ['a', 'b', 'c'])
         exec_insert_table(
-            t.where("add_one(b) <= 3").select("a, b + 1, add(a + 1, subtract_one(c)) + 2, "
-                                              "add(add_one(a), 1L)"),
+            t.where(E.call('add_one', t.b) <= 3)
+             .select("a, b + 1, add(a + 1, subtract_one(c)) + 2, " "add(add_one(a), 1L)"),
```
Comment (WeiZhong94) [inline flink-python/pyflink/table/tests/test_pandas_udf.py:63]: combine to one string?

File: flink-python/pyflink/table/tests/test_window.py [hunk 2]
```diff
@@ -40,8 +41,8 @@ class BatchTableWindowTests(PyFlinkBatchTableTestCase):
 
     def test_tumble_window(self):
         t = self.t_env.from_elements([(1, 1, "Hello")], ["a", "b", "c"])
-        result = t.window(Tumble.over("2.rows").on("a").alias("w"))\
-            .group_by("w, c").select("b.sum")
+        result = t.window(Tumble.over(E.lit(2).rows).on("a").alias("w"))\
```
Comment (WeiZhong94) [inline flink-python/pyflink/table/tests/test_window.py:44]: ".rows" seems not supported. Maybe we should support it or use "row_interval()" instead?

**Review by WeiZhong94 — COMMENTED** [review]
@dianfu Thanks for your PR! Looks good overall, just some minor comments.

**Comment by dianfu** [comment]
@WeiZhong94 Thanks for the review. Updated the PR.

**Review by WeiZhong94 — APPROVED** [review]

File: flink-python/pyflink/table/tests/test_dependency.py [hunk 3]
```diff
@@ -112,7 +113,7 @@ def check_requirements(i):
             ['a', 'b'], [DataTypes.BIGINT(), DataTypes.BIGINT()])
         self.t_env.register_table_sink("Results", table_sink)
         t = self.t_env.from_elements([(1, 2), (2, 5), (3, 1)], ['a', 'b'])
-        exec_insert_table(t.select("check_requirements(a), a"), "Results")
+        exec_insert_table(t.select(E.call('check_requirements', E.col('a')), E.col('a')), "Results")
```
Comment (sunjincheng121) [inline flink-python/pyflink/table/tests/test_dependency.py:116]: `E.col('a')` -> `t.a`

**Review by sunjincheng121 — COMMENTED** [review]
Thanks for the PR @dianfu 

I have only left only a few comments ,and would be great that we develop a usage principle for "E.col ('a ') and t.a" I recommend using ` T.A` :) 

What do you think?

Best,
Jincheng

File: flink-python/pyflink/table/tests/test_correlate.py [hunk 4]
```diff
@@ -39,7 +40,8 @@ def test_join_lateral_with_join_predicate(self):
                                                     "org.apache.flink.table.utils.TableFunc1")
         source = t_env.from_elements([("1", "1#3#5#7"), ("2", "2#4#6#8")], ["id", "words"])
 
-        result = source.join_lateral("split(words) as (word)", "id = word")
+        result = source.join_lateral(E.call('split', source.words).alias('word'),
+                                     E.col('id') == E.col('word'))
```
[the remark this answers is not in the document]
Comment (dianfu) [reply flink-python/pyflink/table/tests/test_correlate.py:44]: **word** is not from the **source** table and so we could not do that.

**Comment by dianfu** [comment]
@sunjincheng121 Thanks for the review. Updated the PR.

File: flink-python/pyflink/table/tests/test_correlate.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sunjincheng121) [reply flink-python/pyflink/table/tests/test_correlate.py]: Yes, I see.

**Comment by sunjincheng121** [comment]
@flinkbot approve all

**Review by sunjincheng121 — APPROVED** [review]

**Pull request merged**